### DISCUSSION TITLE
1:1 cannot create inverse side entity

### DISCUSSION THREAD

**Discussion by Kazma830124** [mikro-orm/mikro-orm#4499] [Q&A] [answered]
**Describe the bug**
In 1:1 relationship ,if  owner primary key as inverse side primary key, using the fieldName/joincolumn decorator options will not create the inverse side

**Stack trace**
```
[query] begin
[query] insert into "test"."advertisers" ("id") values (default) returning "id" [took 2 ms]
[query] commit
Advertisers {
  id: 45,
  quota: Ref<AdvertisersQuota<45>> {
    entity: (AdvertisersQuota) { advertisersId: 45, advertisers: [Ref<Advertisers<45>>] }
  }
}
[query] select "a0".* from "test"."advertisers" as "a0" left join "test"."advertisers_quota" as "a1" on "a0"."id" = "a1"."advertisers_id" where "a1"."video_quota" = 0 [took 2 ms, 13 results]
```

**To Reproduce**
minimal reproduction:
```ts
import { Entity, PrimaryKey, Property, OneToOne, EntityManager, OptionalProps, Ref, BaseEntity } from '@mikro-orm/core';
import { MikroORM } from '@mikro-orm/postgresql';

@Entity()
class AdvertisersQuota extends BaseEntity<AdvertisersQuota,'advertisersId'>{
  [OptionalProps]?:'videoQuota' | 'advertisers'
  @PrimaryKey()
  advertisersId!:string

  @OneToOne({mappedBy: 'quota'})
  advertisers!: Ref<Advertisers>;

  @Property()
  videoQuota: number = 0;
}

@Entity()
class Advertisers  extends BaseEntity<Advertisers,'id'> {
  [OptionalProps]?:'quota'
  @PrimaryKey()
  id!: string;

  @OneToOne({inversedBy: 'advertisers',joinColumn:'id'})
  quota!: Ref<AdvertisersQuota>;
}

async function run() {
  const orm = await MikroORM.init();

  const em = orm.em.fork();
  const advertisers = em.create(Advertisers, {});
  await em.flush()
  
  console.log(advertisers)
  const quota = em.create(AdvertisersQuota, {
    advertisersId:advertisers.id,
  });
  await em.flush()

  await em.find(Advertisers,{quota:{videoQuota:0}})

  await orm.close();
}

run();

```

**Expected behavior**
also create advertiserQuota entity

**Additional context**
if I add ```em.clear() ``` before ```console.log(advertisers)```, the inverse side will create,
remove joincolumn options will create, also.

**Versions**

| Dependency | Version |
| - | - |
| node |v18.14.0 |
| typescript |5.1.6 |
| mikro-orm | 5.7.12 |
| postgres | 15 |

**Comment by B4nan** [comment, marked as answer]
You can't define multiple persistent properties mapping to the same column. Moreover, you cannot have a PK that is inverse side, which is what your mapping currently look like - inverse side of 1:1 relation does not represent a db column. 

Can you provide the schema for what you want to create the entities? I am not entirely sure what you are up to, if you want to share the PK value between the two tables, you can do this:

```ts
@Entity()
class AdvertisersQuota {
  // only one property for this, and it needs to be owner
  @OneToOne({ primary: true })
  advertisers!: Ref<Advertisers>;

  @Property()
  videoQuota: number = 0;
}

@Entity()
class Advertisers {
  // this is the PK that will be shared for both entities thanks to the 1:1 owner PK
  @PrimaryKey()
  id!: string;

  // this is inverse side, only represents a link between the two, it does not represent any column - the link goes to the `AdvertisersQuota.advertisers` which holds the identity of that entity - and shares the PK with this one
  @OneToOne({ mappedBy: 'advertisers' })
  quota!: Ref<AdvertisersQuota>;
}
```

Note that this will work only with ts-morph metadata provider, otherwise you are missing a few more options.

**Reply by Kazma830124** [reply]
Thank for responding and reminding, actually I lack the ts-morph option.

As you mention that it's a mistake I misunderstand owner side and inverse side, 
after I change that and remove the property which defined multiple persistent, now it look like what I expected.
